design/FeaturesWithLimits.md

```diff
@@ -0,0 +1,124 @@
+# Features that add new limits
+
+Some new features add associated limits. This document records a design policy for such features.
+
+## Common behaviors
+
+- `{ newLimit: undefined }` is allowed even if `newLimit` is unknown. (It has no effect.)
+- Limits entries associated with features have these Adapter Capability Guarantees:
+  - They are `>= default` if the feature is available on the adapter.
+  - They are `undefined` if the feature is unavailable on the adapter.
+- It is allowed to request a limit that's available on the adapter (that is,
+  not `undefined` in `adapter.limits`), even without also requesting the corresponding feature.
+  (It won't have any effect.)
```

Comment:
This sentence doesn't make it clear if the limit requested without the feature will still be validated against the adapter-supported limit. I would say yes, since that both simplifies the validation logic and encourages devs to _always_ look at the adapter limits instead of simply plugging in aspirational numbers.